resources/views/pages/claims/completed.blade.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+use App\Models\AchievementSetClaim;
+use App\Platform\Services\AchievementSetClaimListService;
+use Illuminate\Support\Carbon;
+use Illuminate\View\View;
+
+use function Laravel\Folio\{middleware, name, render};
+
+middleware(['auth', 'can:viewAny,' . App\Models\AchievementSetClaim::class]);
```

Comment:
To ensure I'm fully understanding your question, it sounds like you're asking if using both the `auth` middleware and a specific policy on the same page is redundant.

Assuming that's the question: yes, I tend to agree that it can be redundant in many scenarios.

In many cases, it seems like relying solely on the policy can be sufficient. With policies we obviously get flexible and nuanced access control, allowing us to define specific conditions under which a user can access a resource.

My perception is the main benefit of the `auth` middleware is its simplicity/clarity. It ensures a given page is strictly available for authenticated users, without the need to check additional conditions on a policy. It is also immediately clear to the reader that the page requires authentication, so the reader themselves doesn't need to dive into what may be complex RBAC for the route.

There could be scenarios where both are needed. For instance, a page might have a list of publicly-viewable entities, but offer additional interactions or information which should only be visible to authenticated users. In such cases, `auth` middleware makes sense to use, while the policy manages finer access control.

Reply:
I'm just responding to:

> this shouldn't be an authenticated page.

My comment suggests that the page should always check the policy. The policy can decide whether or not the page needs to be authenticated. That way, if we want to change the policy in the future, we only have to change the policy, and won't have to update all the pages.

Reply:
I agree with all of that. I think what I'm saying is if we're to leave it at the sole discretion of the policy, this implies the `auth` middleware must be removed. The `auth` middleware will enforce that `$user` cannot be null, no matter what the policy says.

Reply:
> The auth middleware will enforce that $user cannot be null, no matter what the policy says.

I guess I'm confused about the interactions. Why'd you have me [make the policy accept null](https://github.com/RetroAchievements/RAWeb/pull/2481#discussion_r1625125706) if it can never be provided null?

Reply:
Only for the sake of consistency with other `viewAny` rules that also do not enforce authentication, eg `GamePolicy::viewAny()`:

```php
public function viewAny(?User $user): bool
{
    return true;
}
```

There are quite a few of these in the app; I assume it is convention.

<img width="308" alt="Screenshot 2024-06-04 at 9 41 49 PM" src="https://github.com/RetroAchievements/RAWeb/assets/3984985/090ff450-ecb7-4d9b-9fdb-92befdcd7cdf">

Reply:
Oh, I see. You're not suggesting removing the middleware entirely, just the `auth` clause. That makes sense.

Reply:
done.